pkg/name/name.go

```diff
@@ -76,7 +76,7 @@ const (
 	PrometheusOperatorComponentName ComponentName = "PrometheusOperator"
 	GrafanaComponentName            ComponentName = "Grafana"
 	KialiComponentName              ComponentName = "Kiali"
-	CNIComponentName                ComponentName = "CNI"
+	CNIComponentName                ComponentName = "Cni"
```

Comment:
Why not all caps? CNI is an initialism.

Reply:
We translate first character to lower case for `ComponentName` when translating `values` to `IstioControlPlaneSpec`, if we update `CNIComponentName` to all caps, then we need to treat it as an exception. see: https://github.com/istio/operator/blob/master/pkg/util/path.go#L60-L67

Reply:
If feature and component name is always camel-cased to match value, would it make more sense to also define the constants here to match?

Reply:
It makes sense, but we need to refactor code about translation from values to IstioControlPlaneSpec.

Reply:
Doesn't that mean just removing `firstCharToLowerCase` from `ToYAMLPath`? All of the should be using the constants defined here.

Reply:
I also thought it was, it's not that easy, because `ToYAMLPath` is general function invoked at couples of places, the path contains not only component name and feature name, also we should not keep operator relevant stuff in `util` package.

Reply:
@ayj thiis PR LGTM.  I do not think we should block on this concern.  @morvencao can you file an issue tracker so we don't lose track of this particular issue and can treat it separately from the rest of the PR?

CHeers
-steve

Reply:
Filed an issue to track this: https://github.com/istio/istio/issues/17472